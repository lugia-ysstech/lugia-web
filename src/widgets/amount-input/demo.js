/**
 *
 * create by ligx
 *
 * @flow
 */
import * as React from 'react';
import Theme from '../theme';
import AmountInput from './index';
import Widget from '../consts/index';
import styled from 'styled-components';
import { getBoxShadow } from '@lugia/theme-utils';

class LimitAmountInput extends React.Component<any, any> {
  constructor(props: any) {
    super(props);
    this.state = { value: '' };
  }

  onChange = ({ newValue: value }: any) => {
    this.setState({ value });
    this.props.onChange({ newValue: value });
  };

  render() {
    return <AmountInput value={this.state.value} onChange={this.onChange} />;
  }
}
class DefaultValueAmountInput extends React.Component<any, any> {
  render() {
    return <AmountInput defaultValue={'123456'} onChange={this.props.onChange} />;
  }
}
const Wrapper = styled.div`
  float: left;
  margin-left: 50px;
`;
export default () => {
  const view = {
    [Widget.AmountInput]: {
      Container: {
        normal: {
          width: 400,
          height: 40,
          border: {
            top: 10,
          },
        },
      },
      AmountInputPrefix: { normal: { fontSize: 14, color: 'blue' } },
      AmountTip: {
        Container: {
          normal: {
            background: {
              color: '#eee',
            },
          },
        },
        TooltipTitle: {
          normal: {
            color: '#4d63ff',
            fontSize: 16,
          },
        },
      },
    },
  };
  const onChange = (cmpName: string) => (value: any) => {};
  return (
    <div>
      <Wrapper>
        <p>formatter AmountInput</p>
        <AmountInput placeholder={'请填写金额'} />
        <p>default AmountInput</p>
        <AmountInput placeholder={'请填写金额'} />
        <p>禁用状态 </p>
        <AmountInput size={'default'} disabled={true} />
      </Wrapper>
      <Wrapper>
        <Theme config={view}>
          <p>small size</p>
          <AmountInput size={'small'} placeholder={'请填写金额'} />
          <p>default size</p>
          <AmountInput placeholder={'请填写金额'} />
          <p>large size</p>
          <AmountInput size={'large'} placeholder={'请填写金额'} disabled={true} />
        </Theme>
      </Wrapper>
      <Wrapper>
        <p>受限Input</p>
        <LimitAmountInput onChange={onChange('limit')} />
        <p>有默认值的 受限Input</p>
        <DefaultValueAmountInput value={'123456'} onChange={onChange('limit')} />
      </Wrapper>
      <Wrapper>
        <p>amountPrefix: '¥' transform: false </p>
        <AmountInput amountPrefix="¥" transform={false} />
        <p>amountPrefix: '$' transform: false </p>
        <AmountInput amountPrefix="$" transform={false} />
      </Wrapper>
    </div>
  );
};
