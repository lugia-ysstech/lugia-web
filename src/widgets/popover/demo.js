/**
 *
 * create by liangguodong on 2018/12/5
 *
 * @flow
 */
import * as React from 'react';
import Button from '../button';
import styled from 'styled-components';
import Popover from './popover';
import Input from '../input/index';
import Widget from '../consts';
import Theme from '../theme';

const buttonWidth = 80;
const Direction = styled(Button)`
  width: ${buttonWidth}px;
`;
const description = 'This is description';
const title = 'This is title';
class InnerCloseDemo extends React.Component<any, any> {
  state = {
    visible: false,
  };

  hide = () => {
    this.setState({
      visible: false,
    });
  };

  handleVisibleChange = visible => {
    this.setState({ visible });
  };

  render() {
    return (
      <Popover
        showClearButton
        placement="top"
        title={title}
        action="click"
        visible={this.state.visible}
        onVisibleChange={this.handleVisibleChange}
        clear={'lugia-icon-reminder_close'}
        description={[<div>{description}</div>]}
        onClearClick={this.hide}
      >
        <Direction type="primary">Click me</Direction>
      </Popover>
    );
  }
}
class HoverAndClickDemo extends React.Component<any, any> {
  state = {
    clicked: false,
    hovered: false,
  };

  hide = () => {
    this.setState({
      clicked: false,
      hovered: false,
    });
  };

  handleHoverChange = visible => {
    const { onVisibleChange } = this.props;
    this.setState({
      hovered: visible,
      clicked: false,
    });
    onVisibleChange && onVisibleChange(visible);
  };

  handleClickChange = visible => {
    this.setState({
      clicked: visible,
      hovered: false,
    });
  };

  render() {
    const hoverContent = <div>This is hover content.</div>;
    const clickContent = <div>This is click content.</div>;
    const config = {
      [Widget.Popover]: {
        PopoverContent: {
          Container: {
            normal: { width: 200, background: { color: 'blue' } },
          },
        },
      },
      [Widget.Button]: { width: buttonWidth },
    };
    return (
      <Theme config={config}>
        <Popover
          description={hoverContent}
          title="This is  hover title"
          action={'hover'}
          placement="topLeft"
          visible={this.state.hovered}
          onVisibleChange={this.handleHoverChange}
        >
          <div>
            <Popover
              description={clickContent}
              title="This is  click title"
              action={'click'}
              placement="topLeft"
              visible={this.state.clicked}
              clear={'lugia-icon-reminder_close'}
              onVisibleChange={this.handleClickChange}
              onClearClick={this.hide}
            >
              <Direction type="primary">Hover and click / 悬停并单击</Direction>
            </Popover>
          </div>
        </Popover>
      </Theme>
    );
  }
}
const Wrapper = styled.div`
  margin: 100px;
`;

export const WrapperDemo = () => {
  const text = 'This is title ';
  const description = 'This is description';
  const config = {
    [Widget.Popover]: {
      PopoverClearIcon: {
        normal: { color: 'purple' },
      },
      PopoverContent: {
        Container: { normal: { background: { color: 'pink' } } },
        TooltipTitle: { normal: { color: 'green' } },
        TooltipDescription: { normal: { color: 'yellow' } },
      },
    },
    [Widget.Button]: {
      Container: {
        normal: {
          width: buttonWidth,
        },
      },
    },
  };
  return (
    <Wrapper>
      <Theme config={config}>
        <div style={{ marginLeft: 80, whiteSpace: 'nowrap' }}>
          <Popover placement="topLeft" title={text} action={'click'} description={description}>
            <Direction type="primary">TL</Direction>
          </Popover>
          <Popover
            placement="top"
            title={text}
            description={[<div>{description}</div>, <div>{description}</div>]}
          >
            <Direction type="primary">Top</Direction>
          </Popover>
          <Popover
            placement="topRight"
            title={text}
            description={[<div>{description}</div>, <div>{description}</div>]}
          >
            <Direction type="primary">TR</Direction>
          </Popover>
        </div>
        <div style={{ width: 45, float: 'left' }}>
          <Popover
            placement="leftTop"
            title={text}
            description={[<div>{description}</div>, <div>{description}</div>]}
          >
            <Direction type="primary">LT</Direction>
          </Popover>
          <Popover
            placement="left"
            title={text}
            description={[<div>{description}</div>, <div>{description}</div>]}
          >
            <Direction type="primary">Left</Direction>
          </Popover>
          <Popover
            placement="leftBottom"
            title={text}
            description={[<div>{description}</div>, <div>{description}</div>]}
          >
            <Direction type="primary">LB</Direction>
          </Popover>
        </div>
        <div style={{ width: 50, marginLeft: 320 }}>
          <Popover
            placement="rightTop"
            title={text}
            description={[<div>{description}</div>, <div>{description}</div>]}
          >
            <Direction type="primary">RT</Direction>
          </Popover>
          <Popover
            placement="right"
            title={text}
            description={[<div>{description}</div>, <div>{description}</div>]}
          >
            <Direction type="primary">Right</Direction>
          </Popover>
          <Popover
            placement="rightBottom"
            title={text}
            description={[<div>{description}</div>, <div>{description}</div>]}
          >
            <Direction type="primary">RB</Direction>
          </Popover>
        </div>
        <div style={{ marginLeft: 80, clear: 'both', whiteSpace: 'nowrap' }}>
          <Popover
            placement="bottomLeft"
            title={text}
            description={[<div>{description}</div>, <div>{description}</div>]}
          >
            <Direction type="primary">BL</Direction>
          </Popover>
          <Popover
            placement="bottom"
            title={text}
            description={[<div>{description}</div>, <div>{description}</div>]}
          >
            <Direction type="primary">Bottom</Direction>
          </Popover>
          <Popover
            placement="bottomRight"
            title={text}
            description={[<div>{description}</div>, <div>{description}</div>]}
          >
            <Direction type="primary">BR</Direction>
          </Popover>
        </div>
      </Theme>
      <br />
      <p />
      <Popover
        title={text}
        action={'focus'}
        placement="bottom"
        description={[<div>{description}</div>, <div>{description}</div>]}
      >
        <Input placeholder={'聚焦弹出'} />
      </Popover>
      <Popover
        title={text}
        action={'hover'}
        placement="bottom"
        description={[<div>{description}</div>, <div>{description}</div>]}
      >
        <Direction type="primary"> 悬停</Direction>
      </Popover>
      <Popover
        title={text}
        action={'click'}
        placement="bottom"
        description={[<div>{description}</div>, <div>{description}</div>]}
      >
        <Direction type="primary">点击</Direction>
      </Popover>
      <br />
    </Wrapper>
  );
};
export default () => {
  return [
    <Popover />,
    <WrapperDemo />,
    <Wrapper>
      <HoverAndClickDemo /> <InnerCloseDemo />
    </Wrapper>,
  ];
};
