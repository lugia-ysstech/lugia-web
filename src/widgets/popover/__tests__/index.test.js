/**
 *
 * create by liangguodong
 *
 * @flow
 */
import React from 'react';
import renderer from 'react-test-renderer';
import { WrapperDemo } from '../demo';
import Popover from '../popover';
import Icon from '../../icon';
import Enzyme, { mount } from 'enzyme';
import 'jest-styled-components';
import Adapter from 'enzyme-adapter-react-16';

Enzyme.configure({ adapter: new Adapter() });

describe('popover Demo', () => {
  it('Component JSON', () => {
    global.svtest = true;
    const render = renderer.create(<WrapperDemo />);
    expect(render.toJSON()).toMatchSnapshot();
    global.svtest = false;
  });
  it('props: title', () => {
    const title = 'this is  title!';
    const component = mount(
      <Popover title={title}>
        <div />
      </Popover>
    );
    expect(component.props().title).toBe(title);
  });
  it('props: description ', () => {
    const description = 'this is  description!';
    const component = mount(
      <Popover description={description}>
        <div />
      </Popover>
    );
    expect(component.props().description).toBe(description);
  });
  it('props: description div', () => {
    const description = <div>{'this is description!'}</div>;
    const component = mount(
      <Popover description={description}>
        <div />
      </Popover>
    );
    expect(component.props().description).toBe(description);
  });
  it('props: content ', () => {
    const content = <div>{'this is content!'}</div>;
    const component = mount(
      <Popover content={content}>
        <div />
      </Popover>
    );
    expect(component.props().content).toBe(content);
  });
  it('props: content div', () => {
    const content = [<div>{'this is content!'}</div>, <div>{'this is content!'}</div>];
    const component = mount(
      <Popover content={content}>
        <div />
      </Popover>
    );
    expect(component.props().content).toBe(content);
  });
  it('props: clear = iconClass ', () => {
    const clear = 'lugia-icon-reminder_close';
    const component = mount(
      <Popover clear={clear}>
        <div />
      </Popover>
    );
    expect(component.props().clear).toBe(clear);
  });
  it('props: clear = icon ', () => {
    const clear = (
      <div>
        <Icon iconClass={'lugia-icon-reminder_close'} />
      </div>
    );
    const component = mount(
      <Popover clear={clear}>
        <div />
      </Popover>
    );
    expect(component.props().clear).toBe(clear);
  });
  const getCmp = (target: any): Object => {
    return target
      .children()
      .at(0)
      .instance();
  };
  it('props: visible ', () => {
    const component = mount(
      <Popover visible={true}>
        <div />
      </Popover>
    );
    expect(getCmp(component).state.visible).toBe(true);
  });

  it('props: defaultVisible ', () => {
    const component = mount(
      <Popover defaultVisible={true}>
        <div />
      </Popover>
    );
    expect(getCmp(component).state.visible).toBe(true);
  });

  it('props: defaultVisible && visible ', () => {
    const component = mount(
      <Popover defaultVisible={true} visible={false}>
        <div />
      </Popover>
    );
    expect(getCmp(component).state.visible).toBe(false);
  });
  const onVisibleChange = (visible: boolean) => {
    onVisibleChange && onVisibleChange(visible);
  };

  it('state: visible', () => {
    const component = mount(
      <Popover>
        <div />
      </Popover>
    );
    expect(getCmp(component).state.visible).toBe(false);
  });

  it('state: visible && onVisibleChange', () => {
    const component = mount(
      <Popover onVisibleChange={onVisibleChange}>
        <div />
      </Popover>
    );
    component.setProps({ visible: true });
    expect(getCmp(component).state.visible).toBe(true);
    component.setProps({ visible: false });
    expect(getCmp(component).state.visible).toBe(false);
    component.setProps({ visible: true });
    expect(getCmp(component).state.visible).toBe(true);
  });

  it('props: limit visible && onVisibleChange', () => {
    const component = mount(
      <Popover onVisibleChange={onVisibleChange} visible={true}>
        <div />
      </Popover>
    );
    getCmp(component).setState({ visible: false });
    expect(getCmp(component).state.visible).toBe(true);
  });

  it('props: limit visible && onVisibleChange', () => {
    const component = mount(
      <Popover onVisibleChange={onVisibleChange} visible={false}>
        <div />
      </Popover>
    );
    getCmp(component).setState({ visible: true });
    expect(getCmp(component).state.visible).toBe(false);
  });

  it('props: onVisibleChange event', async () => {
    let onVisibleChange = () => true;
    const result = [];
    const onVisibleChangePromise = new Promise(res => {
      onVisibleChange = visbile => {
        result.push(visbile);
        if (result.length === 2) {
          res(result);
        }
      };
    });
    const component = mount(
      <Popover onVisibleChange={onVisibleChange}>
        <input className={'input'} />
      </Popover>
    );
    const input = component.find('.input');
    input.simulate('click', {});
    input.simulate('click', {});
    expect(await onVisibleChangePromise).toEqual([true, false]);
  });
});
