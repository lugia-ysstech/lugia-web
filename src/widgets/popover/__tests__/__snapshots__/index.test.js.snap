// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`popover Demo Component JSON 1`] = `
.c1 {
  display: inline-block;
  margin-bottom: 0;
  box-sizing: border-box;
  text-align: center;
  touch-action: manipulation;
  cursor: pointer;
  white-space: nowrap;
  font-family: 'Trebuchet MS',Arial,Helvetica,sans-serif;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  -webkit-transition: all 0.3s cubic-bezier(0.645,0.045,0.355,1);
  transition: all 0.3s cubic-bezier(0.645,0.045,0.355,1);
  position: relative;
  text-transform: none;
  outline: 0;
  background-color: #ffffff;
  border-top-width: 0.1rem;
  border-top-style: solid;
  border-top-color: #cccccc;
  border-bottom-width: 0.1rem;
  border-bottom-style: solid;
  border-bottom-color: #cccccc;
  border-left-width: 0.1rem;
  border-left-style: solid;
  border-left-color: #cccccc;
  border-right-width: 0.1rem;
  border-right-style: solid;
  border-right-color: #cccccc;
}

.c2 {
  color: #666666;
  font-size: 1.4rem;
}

.c4 {
  line-height: 1.5;
  display: inline-block;
  font-family: inherit;
  -webkit-transition: all 0.3s;
  transition: all 0.3s;
  outline: none;
  width: 20rem;
  height: 3.2rem;
  cursor: text;
  border-top-width: 0.1rem;
  border-top-style: solid;
  border-top-color: #cccccc;
  border-bottom-width: 0.1rem;
  border-bottom-style: solid;
  border-bottom-color: #cccccc;
  border-left-width: 0.1rem;
  border-left-style: solid;
  border-left-color: #cccccc;
  border-right-width: 0.1rem;
  border-right-style: solid;
  border-right-color: #cccccc;
  border-top-left-radius: 0.4rem;
  border-top-right-radius: 0.4rem;
  border-bottom-right-radius: 0.4rem;
  border-bottom-left-radius: 0.4rem;
}

.c4::-webkit-input-placeholder {
  color: #cccccc;
}

.c4::-moz-placeholder {
  color: #cccccc;
}

.c4:-ms-input-placeholder {
  color: #cccccc;
}

.c4::placeholder {
  color: #cccccc;
}

.c3 {
  position: relative;
  display: inline-block;
  outline: none;
}

.c0 {
  margin: 100px;
}

<div
  className="c0"
>
  <div
    style={
      Object {
        "marginLeft": 80,
        "whiteSpace": "nowrap",
      }
    }
  >
    <button
      className="c1"
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onMouseDown={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      onMouseOut={[Function]}
      onMouseOver={[Function]}
      onMouseUp={[Function]}
      size="default"
      style={
        Object {
          "backgroundColor": "#4d63ff",
          "borderBottomLeftRadius": "0.4rem",
          "borderBottomRightRadius": "0.4rem",
          "borderBottomWidth": 0,
          "borderLeftWidth": 0,
          "borderRightWidth": 0,
          "borderTopLeftRadius": "0.4rem",
          "borderTopRightRadius": "0.4rem",
          "borderTopWidth": 0,
          "height": "3.2rem",
          "paddingBottom": "0rem",
          "paddingLeft": "1.8rem",
          "paddingRight": "1.8rem",
          "paddingTop": "0rem",
          "width": "8rem",
        }
      }
      type="primary"
    >
      <span
        className="c2"
        style={
          Object {
            "color": "#ffffff",
            "fontSize": "1.4rem",
          }
        }
      >
        TL
      </span>
    </button>
    <button
      className="c1"
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onMouseDown={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      onMouseOut={[Function]}
      onMouseOver={[Function]}
      onMouseUp={[Function]}
      size="default"
      style={
        Object {
          "backgroundColor": "#4d63ff",
          "borderBottomLeftRadius": "0.4rem",
          "borderBottomRightRadius": "0.4rem",
          "borderBottomWidth": 0,
          "borderLeftWidth": 0,
          "borderRightWidth": 0,
          "borderTopLeftRadius": "0.4rem",
          "borderTopRightRadius": "0.4rem",
          "borderTopWidth": 0,
          "height": "3.2rem",
          "paddingBottom": "0rem",
          "paddingLeft": "1.8rem",
          "paddingRight": "1.8rem",
          "paddingTop": "0rem",
          "width": "8rem",
        }
      }
      type="primary"
    >
      <span
        className="c2"
        style={
          Object {
            "color": "#ffffff",
            "fontSize": "1.4rem",
          }
        }
      >
        Top
      </span>
    </button>
    <button
      className="c1"
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onMouseDown={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      onMouseOut={[Function]}
      onMouseOver={[Function]}
      onMouseUp={[Function]}
      size="default"
      style={
        Object {
          "backgroundColor": "#4d63ff",
          "borderBottomLeftRadius": "0.4rem",
          "borderBottomRightRadius": "0.4rem",
          "borderBottomWidth": 0,
          "borderLeftWidth": 0,
          "borderRightWidth": 0,
          "borderTopLeftRadius": "0.4rem",
          "borderTopRightRadius": "0.4rem",
          "borderTopWidth": 0,
          "height": "3.2rem",
          "paddingBottom": "0rem",
          "paddingLeft": "1.8rem",
          "paddingRight": "1.8rem",
          "paddingTop": "0rem",
          "width": "8rem",
        }
      }
      type="primary"
    >
      <span
        className="c2"
        style={
          Object {
            "color": "#ffffff",
            "fontSize": "1.4rem",
          }
        }
      >
        TR
      </span>
    </button>
  </div>
  <div
    style={
      Object {
        "float": "left",
        "width": 45,
      }
    }
  >
    <button
      className="c1"
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onMouseDown={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      onMouseOut={[Function]}
      onMouseOver={[Function]}
      onMouseUp={[Function]}
      size="default"
      style={
        Object {
          "backgroundColor": "#4d63ff",
          "borderBottomLeftRadius": "0.4rem",
          "borderBottomRightRadius": "0.4rem",
          "borderBottomWidth": 0,
          "borderLeftWidth": 0,
          "borderRightWidth": 0,
          "borderTopLeftRadius": "0.4rem",
          "borderTopRightRadius": "0.4rem",
          "borderTopWidth": 0,
          "height": "3.2rem",
          "paddingBottom": "0rem",
          "paddingLeft": "1.8rem",
          "paddingRight": "1.8rem",
          "paddingTop": "0rem",
          "width": "8rem",
        }
      }
      type="primary"
    >
      <span
        className="c2"
        style={
          Object {
            "color": "#ffffff",
            "fontSize": "1.4rem",
          }
        }
      >
        LT
      </span>
    </button>
    <button
      className="c1"
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onMouseDown={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      onMouseOut={[Function]}
      onMouseOver={[Function]}
      onMouseUp={[Function]}
      size="default"
      style={
        Object {
          "backgroundColor": "#4d63ff",
          "borderBottomLeftRadius": "0.4rem",
          "borderBottomRightRadius": "0.4rem",
          "borderBottomWidth": 0,
          "borderLeftWidth": 0,
          "borderRightWidth": 0,
          "borderTopLeftRadius": "0.4rem",
          "borderTopRightRadius": "0.4rem",
          "borderTopWidth": 0,
          "height": "3.2rem",
          "paddingBottom": "0rem",
          "paddingLeft": "1.8rem",
          "paddingRight": "1.8rem",
          "paddingTop": "0rem",
          "width": "8rem",
        }
      }
      type="primary"
    >
      <span
        className="c2"
        style={
          Object {
            "color": "#ffffff",
            "fontSize": "1.4rem",
          }
        }
      >
        Left
      </span>
    </button>
    <button
      className="c1"
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onMouseDown={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      onMouseOut={[Function]}
      onMouseOver={[Function]}
      onMouseUp={[Function]}
      size="default"
      style={
        Object {
          "backgroundColor": "#4d63ff",
          "borderBottomLeftRadius": "0.4rem",
          "borderBottomRightRadius": "0.4rem",
          "borderBottomWidth": 0,
          "borderLeftWidth": 0,
          "borderRightWidth": 0,
          "borderTopLeftRadius": "0.4rem",
          "borderTopRightRadius": "0.4rem",
          "borderTopWidth": 0,
          "height": "3.2rem",
          "paddingBottom": "0rem",
          "paddingLeft": "1.8rem",
          "paddingRight": "1.8rem",
          "paddingTop": "0rem",
          "width": "8rem",
        }
      }
      type="primary"
    >
      <span
        className="c2"
        style={
          Object {
            "color": "#ffffff",
            "fontSize": "1.4rem",
          }
        }
      >
        LB
      </span>
    </button>
  </div>
  <div
    style={
      Object {
        "marginLeft": 320,
        "width": 50,
      }
    }
  >
    <button
      className="c1"
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onMouseDown={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      onMouseOut={[Function]}
      onMouseOver={[Function]}
      onMouseUp={[Function]}
      size="default"
      style={
        Object {
          "backgroundColor": "#4d63ff",
          "borderBottomLeftRadius": "0.4rem",
          "borderBottomRightRadius": "0.4rem",
          "borderBottomWidth": 0,
          "borderLeftWidth": 0,
          "borderRightWidth": 0,
          "borderTopLeftRadius": "0.4rem",
          "borderTopRightRadius": "0.4rem",
          "borderTopWidth": 0,
          "height": "3.2rem",
          "paddingBottom": "0rem",
          "paddingLeft": "1.8rem",
          "paddingRight": "1.8rem",
          "paddingTop": "0rem",
          "width": "8rem",
        }
      }
      type="primary"
    >
      <span
        className="c2"
        style={
          Object {
            "color": "#ffffff",
            "fontSize": "1.4rem",
          }
        }
      >
        RT
      </span>
    </button>
    <button
      className="c1"
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onMouseDown={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      onMouseOut={[Function]}
      onMouseOver={[Function]}
      onMouseUp={[Function]}
      size="default"
      style={
        Object {
          "backgroundColor": "#4d63ff",
          "borderBottomLeftRadius": "0.4rem",
          "borderBottomRightRadius": "0.4rem",
          "borderBottomWidth": 0,
          "borderLeftWidth": 0,
          "borderRightWidth": 0,
          "borderTopLeftRadius": "0.4rem",
          "borderTopRightRadius": "0.4rem",
          "borderTopWidth": 0,
          "height": "3.2rem",
          "paddingBottom": "0rem",
          "paddingLeft": "1.8rem",
          "paddingRight": "1.8rem",
          "paddingTop": "0rem",
          "width": "8rem",
        }
      }
      type="primary"
    >
      <span
        className="c2"
        style={
          Object {
            "color": "#ffffff",
            "fontSize": "1.4rem",
          }
        }
      >
        Right
      </span>
    </button>
    <button
      className="c1"
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onMouseDown={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      onMouseOut={[Function]}
      onMouseOver={[Function]}
      onMouseUp={[Function]}
      size="default"
      style={
        Object {
          "backgroundColor": "#4d63ff",
          "borderBottomLeftRadius": "0.4rem",
          "borderBottomRightRadius": "0.4rem",
          "borderBottomWidth": 0,
          "borderLeftWidth": 0,
          "borderRightWidth": 0,
          "borderTopLeftRadius": "0.4rem",
          "borderTopRightRadius": "0.4rem",
          "borderTopWidth": 0,
          "height": "3.2rem",
          "paddingBottom": "0rem",
          "paddingLeft": "1.8rem",
          "paddingRight": "1.8rem",
          "paddingTop": "0rem",
          "width": "8rem",
        }
      }
      type="primary"
    >
      <span
        className="c2"
        style={
          Object {
            "color": "#ffffff",
            "fontSize": "1.4rem",
          }
        }
      >
        RB
      </span>
    </button>
  </div>
  <div
    style={
      Object {
        "clear": "both",
        "marginLeft": 80,
        "whiteSpace": "nowrap",
      }
    }
  >
    <button
      className="c1"
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onMouseDown={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      onMouseOut={[Function]}
      onMouseOver={[Function]}
      onMouseUp={[Function]}
      size="default"
      style={
        Object {
          "backgroundColor": "#4d63ff",
          "borderBottomLeftRadius": "0.4rem",
          "borderBottomRightRadius": "0.4rem",
          "borderBottomWidth": 0,
          "borderLeftWidth": 0,
          "borderRightWidth": 0,
          "borderTopLeftRadius": "0.4rem",
          "borderTopRightRadius": "0.4rem",
          "borderTopWidth": 0,
          "height": "3.2rem",
          "paddingBottom": "0rem",
          "paddingLeft": "1.8rem",
          "paddingRight": "1.8rem",
          "paddingTop": "0rem",
          "width": "8rem",
        }
      }
      type="primary"
    >
      <span
        className="c2"
        style={
          Object {
            "color": "#ffffff",
            "fontSize": "1.4rem",
          }
        }
      >
        BL
      </span>
    </button>
    <button
      className="c1"
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onMouseDown={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      onMouseOut={[Function]}
      onMouseOver={[Function]}
      onMouseUp={[Function]}
      size="default"
      style={
        Object {
          "backgroundColor": "#4d63ff",
          "borderBottomLeftRadius": "0.4rem",
          "borderBottomRightRadius": "0.4rem",
          "borderBottomWidth": 0,
          "borderLeftWidth": 0,
          "borderRightWidth": 0,
          "borderTopLeftRadius": "0.4rem",
          "borderTopRightRadius": "0.4rem",
          "borderTopWidth": 0,
          "height": "3.2rem",
          "paddingBottom": "0rem",
          "paddingLeft": "1.8rem",
          "paddingRight": "1.8rem",
          "paddingTop": "0rem",
          "width": "8rem",
        }
      }
      type="primary"
    >
      <span
        className="c2"
        style={
          Object {
            "color": "#ffffff",
            "fontSize": "1.4rem",
          }
        }
      >
        Bottom
      </span>
    </button>
    <button
      className="c1"
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onMouseDown={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      onMouseOut={[Function]}
      onMouseOver={[Function]}
      onMouseUp={[Function]}
      size="default"
      style={
        Object {
          "backgroundColor": "#4d63ff",
          "borderBottomLeftRadius": "0.4rem",
          "borderBottomRightRadius": "0.4rem",
          "borderBottomWidth": 0,
          "borderLeftWidth": 0,
          "borderRightWidth": 0,
          "borderTopLeftRadius": "0.4rem",
          "borderTopRightRadius": "0.4rem",
          "borderTopWidth": 0,
          "height": "3.2rem",
          "paddingBottom": "0rem",
          "paddingLeft": "1.8rem",
          "paddingRight": "1.8rem",
          "paddingTop": "0rem",
          "width": "8rem",
        }
      }
      type="primary"
    >
      <span
        className="c2"
        style={
          Object {
            "color": "#ffffff",
            "fontSize": "1.4rem",
          }
        }
      >
        BR
      </span>
    </button>
  </div>
  <br />
  <p />
  <div
    className="c3"
    onMouseDown={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    style={Object {}}
  >
    <input
      autoFocus={false}
      className="c4"
      disabled={false}
      onBlur={[Function]}
      onChange={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyPress={[Function]}
      onKeyUp={[Function]}
      placeholder="聚焦弹出"
      size="default"
      style={
        Object {
          "color": "#333333",
          "height": "3.2rem",
          "paddingBottom": "0rem",
          "paddingLeft": "1rem",
          "paddingRight": "3.5rem",
          "paddingTop": "0rem",
        }
      }
      value=""
    />
  </div>
  <button
    className="c1"
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onMouseDown={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    onMouseOut={[Function]}
    onMouseOver={[Function]}
    onMouseUp={[Function]}
    size="default"
    style={
      Object {
        "backgroundColor": "#4d63ff",
        "borderBottomLeftRadius": "0.4rem",
        "borderBottomRightRadius": "0.4rem",
        "borderBottomWidth": 0,
        "borderLeftWidth": 0,
        "borderRightWidth": 0,
        "borderTopLeftRadius": "0.4rem",
        "borderTopRightRadius": "0.4rem",
        "borderTopWidth": 0,
        "height": "3.2rem",
        "paddingBottom": "0rem",
        "paddingLeft": "1.8rem",
        "paddingRight": "1.8rem",
        "paddingTop": "0rem",
      }
    }
    type="primary"
  >
    <span
      className="c2"
      style={
        Object {
          "color": "#ffffff",
          "fontSize": "1.4rem",
        }
      }
    >
       悬停
    </span>
  </button>
  <button
    className="c1"
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onMouseDown={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    onMouseOut={[Function]}
    onMouseOver={[Function]}
    onMouseUp={[Function]}
    size="default"
    style={
      Object {
        "backgroundColor": "#4d63ff",
        "borderBottomLeftRadius": "0.4rem",
        "borderBottomRightRadius": "0.4rem",
        "borderBottomWidth": 0,
        "borderLeftWidth": 0,
        "borderRightWidth": 0,
        "borderTopLeftRadius": "0.4rem",
        "borderTopRightRadius": "0.4rem",
        "borderTopWidth": 0,
        "height": "3.2rem",
        "paddingBottom": "0rem",
        "paddingLeft": "1.8rem",
        "paddingRight": "1.8rem",
        "paddingTop": "0rem",
      }
    }
    type="primary"
  >
    <span
      className="c2"
      style={
        Object {
          "color": "#ffffff",
          "fontSize": "1.4rem",
        }
      }
    >
      点击
    </span>
  </button>
  <br />
</div>
`;
